sklearn/metrics/scorer.py

```diff
@@ -135,7 +135,10 @@ def __call__(self, clf, X, y, sample_weight=None):
         """
         super(_ProbaScorer, self).__call__(clf, X, y,
                                            sample_weight=sample_weight)
+        y_type = type_of_target(y)
         y_pred = clf.predict_proba(X)
+        if y_type == "binary":
+            y_pred = y_pred[:, 1]
```

Comment:
@jnothman Sorry but I don't quite understand your meaning. Currently, it seems that we don't test the wrapper of the scorer. Instead, we directly test the scorer to ensure a reasonable behaviour. And I have added the new scorer to the test.